components/server-side-render/index.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: {},
+			attributes: props,
+		};
+	}
+
+	componentDidMount() {
+		this.getOutput();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( JSON.stringify( nextProps ) !== JSON.stringify( this.props ) ) {
+			this.setState( { attributes: nextProps }, this.getOutput );
+		}
+	}
+
+	getOutput() {
+		this.setState( { response: {} } );
+		const { block } = this.props;
+		const attributes = this.state.attributes;
+		const apiURL = addQueryArgs( '/gutenberg/v1/blocks-renderer/' + block, {
+			...attributes,
+			_wpnonce: wpApiSettings.nonce,
+		} );
+
+		return wp.apiRequest( { path: apiURL } ).then( response => {
+			if ( response && response.output ) {
+				this.setState( { response: response.output } );
+			}
+		} );
+	}
+
+	render() {
+		const response = this.state.response;
+		if ( ! response.length ) {
+			return (
+				<div key="loading" className="wp-block-embed is-loading">
+
+					<p>{ __( 'Loading...' ) }</p>
+				</div>
+			);
+		}
+
+		return (
+			<RawHTML key="html">{ response }</RawHTML>
```

Comment:
Should this use `SandBox` instead?

Reply:
I guess not because then styles won't be applied.

Reply:
Sandbox doesn't stop styles being applied. It's used in #4710. Should work on such block-specifics defer to this once it's released?

Reply:
I mean the CSS from the editor. Otherwise, the CSS has to be re-printed in the iframe.